lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3281,6 +3281,17 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       // Record the body.
       TC.definedFunctions.push_back(FD);
     }
+    
+    // If any of the function's parameters contain an uninhabited type
+    // then emit a diagnostic on the func decl, as the function can never
+    // be called.
+    auto params = FD->getParameters()->getArray();
+    for (auto &&index: indices(params)) {
```

Comment:
Nitpick: You're not using 'index' anywhere in the body, so you can just iterate directly over params:

`for (auto param : *FD->getParameters())`

Reply:
Fixed!